trim.production.js

```diff
@@ -31,16 +31,24 @@ function createRecord (title, container, extension, fileData, alternativeContain
     throw new Error('alternativeContainers must be a list of strings: ' + alternativeContainers)
   }
 
+  var body = {
+    Title: title,
+    Container: container,
+    RecordExtension: extension,
+    AlternativeContainers: alternativeContainers,
+    Record: fileData
+  }
+
+  body = R.evolve({
+    Title: R.ifElse(R.isNilOrEmpty, R.always('Untitled Document'), R.replace(/:/g, '-')),
+    RecordExtension: R.replace(/./g, ''),
```

Comment:
It's just to remove any periods in the extension in case it gets left in.
`R.replace` just delegates to `'the string'.replace` anyways I think. How
would you do it?

On Tuesday, January 19, 2016, Mark Stuart <notifications@github.com> wrote:

> In trim.production.js
> <https://github.com/mediasuitenz/trimperson/pull/9#discussion_r50076862>:
>
> > @@ -31,16 +31,24 @@ function createRecord (title, container, extension, fileData, alternativeContain
> >      throw new Error('alternativeContainers must be a list of strings: ' + alternativeContainers)
> >    }
> >
> > +  var body = {
> > +    Title: title,
> > +    Container: container,
> > +    RecordExtension: extension,
> > +    AlternativeContainers: alternativeContainers,
> > +    Record: fileData
> > +  }
> > +
> > +  body = R.evolve({
> > +    Title: R.ifElse(R.isNilOrEmpty, R.always('Untitled Document'), R.replace(/:/g, '-')),
> > +    RecordExtension: R.replace(/./g, ''),
>
> Slightly strange way of setting the value to an empty string? At least I
> think that's what it's doing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mediasuitenz/trimperson/pull/9/files#r50076862>.
>